text/3-entrees/fixed-point.md

```diff
@@ -0,0 +1,329 @@
+	let old_accumulated : u64 = Self::manual_value() as u64;
+	let new_factor_u64 : u64 = new_factor as u64;
+
+	// Perform the multiplication on the u64 values
+	// This intermediate format has 32 integer positions and 32 fractional.
+	let raw_product : u64 = old_accumulated * new_factor_u64;
+
+	// Right shift to restore the convention that 16 bits are fractional.
+	// This is a lossy conversion.
+	// This intermediate format has 48 integer positions and 16 fractional.
+	let shifted_product : u64 = raw_product >> 16;
+
+	// Ensure that the product fits in the u32, and error if it doesn't
+	if shifted_product > (u32::max_value() as u64) {
+		return Err(Error::<T>::Overflow.into())
+	}
+
+	// Write the new value to storage
+	ManualAccumulator::put(shifted_product as u32);
+
+	// Emit event
+	Self::deposit_event(Event::ManualUpdated(new_factor, shifted_product as u32));
+	Ok(())
+}
+```
+
+As mentioned above, when you multiply two 32-bit numbers, you can end up with as much as 64 bits in the product. In this 64-bit intermediate product, we have 32 integer bits and 32 fractional. We can simply throw away the 16 right-most fractional bits that merely provide extra precision. But we need to be careful with the 16 left-most integer bits. If any of those bits are non-zero after the multiplication it means overflow has occurred. If they are all zero, then we can safely throw them away as well.
+
+> If this business about having more bits after the multiplication is confusing, try this exercise in the more familiar decimal system. Consider these numbers that have 4 total digits (2 integer, and two fractional): 12.34 and 56.78. Multiply them together. How many integer and fractional digits are in the product? Try that again with larger numbers like 01.23 * 02.48 and smaller like 11.11 and 22.22. Which of these products can be fit back into a 4-digit number like the ones we started with?
+
+## Compounding Interest
+*[`pallets/compounding-interest`](https://github.com/substrate-developer-hub/recipes/tree/master/pallets/compounding-interest)*
+
+Many financial agreements involve interest for loaned or borrowed money. [Compounding interest](https://en.wikipedia.org/wiki/Compound_interest) is when new interest is paid on top of not only the original loan amount, the so-called "principle", but also any interest that has been previously paid.
+
+### Discrete Compounding
+
+Our first example will look at discrete compounding interest. This is when interest is paid at a fixed interval. In our case, interest will be paid every ten blocks.
+
+For this implementation we've chosen to use Substrate's [`Percent` type](https://substrate.dev/rustdocs/master/sp_arithmetic/struct.Percent.html). It works nearly the same as `Permill`, but it represents numbers as "parts per hundred" rather than "parts er million". We could also have used Substrate-fixed for this implementation, but chose to save it for the next example.
```

Comment:
```suggestion
For this implementation we've chosen to use Substrate's [`Percent` type](https://substrate.dev/rustdocs/master/sp_arithmetic/struct.Percent.html). It works nearly the same as `Permill`, but it represents numbers as "parts per hundred" rather than "parts per million". We could also have used Substrate-fixed for this implementation, but chose to save it for the next example.
```